packages/devextreme/js/__internal/ui/overlay/m_overlay_position_controller.ts

```diff
@@ -1,101 +1,163 @@
+import type { PositionAlignment } from '@js/common';
 import positionUtils from '@js/common/core/animation/position';
 import { locate, move, resetPosition } from '@js/common/core/animation/translator';
-import type { dxElementWrapper } from '@js/core/renderer';
+import type { Coordinates, dxElementWrapper } from '@js/core/renderer';
 import $ from '@js/core/renderer';
 import { extend } from '@js/core/utils/extend';
 import {
   isDefined,
-  // @ts-expect-error private prop
   isEvent,
   isString,
   isWindow,
 } from '@js/core/utils/type';
 import swatch from '@js/ui/widget/swatch_container';
+import type {
+  OverlayActions,
+  OverlayProperties,
+  PointerLikeEvent,
+} from '@ts/ui/overlay/overlay';
 
 import windowUtils from '../../core/utils/m_window';
 
-interface OverlayPositionControllerConstructor {
-  position?: unknown;
-  container?: unknown;
-  visualContainer?: unknown;
+export type OverlayPositionAlignment =
+  | 'top center'
+  | 'bottom center'
+  | 'right center'
+  | 'left center'
+  | 'center'
+  | 'right bottom'
+  | 'right top'
+  | 'left bottom'
+  | 'left top';
+
+export interface OverlayPosition extends Partial<Coordinates> {
+  my?: OverlayPositionAlignment;
+  at?: OverlayPositionAlignment;
+  of?: string | dxElementWrapper | Element;
+  offset?: string | { x?: number; y?: number };
+  boundaryOffset?: typeof DEFAULT_BOUNDARY_OFFSET;
+}
+
+export type Position = OverlayPosition | OverlayPositionAlignment;
+
+export interface ControllerOverlayElements {
   $root?: dxElementWrapper;
   $content?: dxElementWrapper;
   $wrapper?: dxElementWrapper;
-  onPositioned?: unknown;
-  onVisualPositionChanged?: unknown;
-  restorePosition?: unknown;
-  _fixWrapperPosition?: unknown;
-  _skipContentPositioning?: unknown;
+}
+
+export interface BaseControllerProperties {
```

Comment:
we can use Pick here for shorter syntax
```suggestion
    BaseControllerProperties = Pick<OverlayProperties, 'container' | 'visualContainer' | 'position' | 'restorePosition' | '_fixWrapperPosition' | '_skipContentPositioning'>
```

Reply:
Good point, will use

Reply:
cc8d60406c7180ce1915a9d9939ff239c48318f7